channeldb/addr_test.go

```diff
@@ -43,7 +44,7 @@ var addrTests = []struct {
 	},
 	{
 		expAddr: unknownAddrType{},
-		serErr:  ErrUnknownAddressType,
+		serErr:  ErrUnknownAddressType.Error(),
```

Comment:
Hmm, this seems like step backwards...do we want to use a proper type instead?

Reply:
In case of invalid addresses, we want the invalid address to be part of the returned error to ease debugging, so we cannot use a static error for these cases. We _could_ make  a wrapping error type (as we do with `ReservationError`) but it seemed a bit overkill just to accommodate testing.

Reply:
PTAL @Roasbeef

Reply:
As an alternative, we could replace `serErr` with a `valid bool` instead. Is there any reason we're interested in the exact error of the serialization?

Reply:
It lets you check that a it fails in the expected way. I guess checking the string is a middle ground between only checkin if any error at all and checking the exact type.